server/db.js

```diff
@@ -81,34 +81,51 @@ const fetchRecipeFromPeachworks = (id, callback) => {
   peachworks.proxyGetInventory(id).then(invJson => {
     let inventory = invJson.map(i => {
       return {
-        //TODO: what is quantity vs common_quantity?
         quantity: parseFloat(i.quantity),
         itemId: i.item_id,
-        unitId: i.unit_id
+        unitId: i.unit_id,
+        customUnitId: i.each_unit_id
       };
     });
 
+    const fetchForInv = (idKey, peachworksApi, invKey, jsonKey) => {
+      const ids = inventory.map(i => i[idKey]).filter(i => i);
+
+      return peachworksApi(ids).then(json => {
+        inventory = inventory.map(i => {
+          if (i[idKey]) {
+            const jsonItem = json.find(j => i[idKey] === j.id);
+            //TODO this is pretty gross... any better way to do this?
+            i[invKey] = jsonItem[jsonKey];
```

Comment:
Maybe instead of doing nested array scans, you could just transform the json response array once to a map keyed on the thing you're interested in comparing for equality?

This should yield be (n_i + n_j) iterations instead of (n_i*n_j) iterations, right?
You pay a memory cost for the auxiliary map creation.

e.g. something like:

```
const jsonMapById = new Map(json.map((j) => [j.id, j]));

inventory.map(i = {
  ...
  const idVal = i[idKey]
  const jsonItem = jsonMapById.get(idVal)
})
```